Makefile does not respect DESTDIR

Hi, I'm the AUR maintainer for this package. I believe @MCotocel had asked earlier if I could start building this with the Makefile instead of directly invoking `go build`, however, the Makefile in question doesn't seem to respect whatever DESTDIR is set to. This is important for Arch Linux, as packages are not allowed to directly install to `/usr/bin` without using `pacman`.

This is what I get from an unreleased hilbish PKGBUILD, using the latest release:
```shell
make: Warning: File 'Makefile' has modification time 5828 s in the future
'hilbish' -> '/usr/bin/hilbish'
install: cannot create regular file '/usr/bin/hilbish': Permission denied
make: *** [Makefile:9: install] Error 1
A failure occurred in package().
Aborting...
```

And this is what I have in the PKGBUILD:
```shell
package() {
  cd "$pkgname-$pkgver"
  DESTDIR="$pkgdir/" make install
```

`DESTDIR` is set to `$pkgdir`, which expands to the package directory. So ideally it should install to `$pkgdir/usr/bin/hilbish`

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ LIBDIR ?= $(PREFIX)/share/hilbish
 build:
 	@go build
 
-install: build
+install:
 	@install -v -d "$(DESTDIR)$(BINDIR)/" && install -m 0755 -v hilbish "$(DESTDIR)$(BINDIR)/hilbish"
 	@mkdir -p "$(DESTDIR)$(LIBDIR)"
 	@cp libs preload.lua .hilbishrc.lua "$(DESTDIR)$(LIBDIR)" -r

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ uninstall:
 	@rm -vrf \
 			"$(DESTDIR)$(BINDIR)/hilbish" \
 			"$(DESTDIR)$(LIBDIR)"
-	@sed -i '/$(DESTDIR)\/usr\/bin\/hilbish/d' /etc/shells
+	@sed -i '/hilbish/d' /etc/shells
 	@echo "Hilbish Uninstalled"
 
 clean:

--- a/Makefile
+++ b/Makefile
@@ -1,22 +1,23 @@
 PREFIX ?= /usr
+DESTDIR ?=
 BINDIR ?= $(PREFIX)/bin
 LIBDIR ?= $(PREFIX)/share/hilbish
 
 build:
 	@go build
 
-install:
-	@install -v -d "$(BINDIR)/" && install -m 0755 -v hilbish "$(BINDIR)/hilbish"
-	@mkdir -p "$(LIBDIR)"
-	@cp libs preload.lua .hilbishrc.lua "$(LIBDIR)" -r
-	@echo /usr/bin/hilbish >> /etc/shells
+install: build
+	@install -v -d "$(DESTDIR)$(BINDIR)/" && install -m 0755 -v hilbish "$(DESTDIR)$(BINDIR)/hilbish"
+	@mkdir -p "$(DESTDIR)$(LIBDIR)"
+	@cp libs preload.lua .hilbishrc.lua "$(DESTDIR)$(LIBDIR)" -r
+	@echo "$(DESTDIR)$(BINDIR)/hilbish" >> /etc/shells
 	@echo "Hilbish Installed"
 
 uninstall:
 	@rm -vrf \
-			"$(BINDIR)/hilbish" \
-			"$(LIBDIR)"
-	@sed '/\/usr\/bin\/hilbish/d' /etc/shells
+			"$(DESTDIR)$(BINDIR)/hilbish" \
+			"$(DESTDIR)$(LIBDIR)"
+	@sed -i '/$(DESTDIR)\/usr\/bin\/hilbish/d' /etc/shells
 	@echo "Hilbish Uninstalled"
 
 clean:

--- a/README.md
+++ b/README.md
@@ -22,7 +22,7 @@ cd Hilbish
 make build
 sudo make install
 # Or 
-sudo make
+sudo make all
 ```
 
 Alternativly, if you use Arch Linux, you can install Hilbish with an **(unofficial)** AUR package

